### PR TITLE
makes s3 bucket domain name style configurable

### LINKED ISSUE #1301: Investigate error with Berks and recent upgrade
When I upgraded 2 of our prod nodes to 2.6.0, we started seeing these errors when running berks install:

Resolving cookbook dependencies with Berkshelf 4.3.3...
E, [2016-05-16T22:20:46.045779 #3130] ERROR -- : Actor crashed!
OpenSSL::SSL::SSLError: hostname "community-files.opscode.com.s3.amazonaws.com" does not match the server certificate

I believe this is due to this PR: #1281

### PR DESCRIPTION
**Pull request by nellshamrell**
Fixes #1301 

Before this is deployed, we will need to add this value to the supermarket vault
s3_domain_style: ':s3_path_url'

### CODE REVIEW

File: src/supermarket/docs/CONFIGURING.md [hunk 1]
```diff
@@ -105,6 +105,9 @@ the test suite.
   cookbook object itself, which may conflict with a bucket policy if you use one
   on your bucket.  If you wish to use a bucket policy, rather than an object policy,
   leave this variable as nil.
+* `S3_DOMAIN_STYLE` is used to specify what type of URL is used to access your s3 bucket.
+  The default is ':s3_domain_url', it will use domains that look like this: bucketname.s3.amazonaws.com.  This works across regions.
```
Comment (robbkidd) [inline src/supermarket/docs/CONFIGURING.md]: Where is the `:s3_domain_url` default declared?

Comment (nellshamrell) [inline src/supermarket/docs/CONFIGURING.md]: And this is why code review is excellent.  Will fix!

**Comment by nellshamrell** [comment]
Although this is setting ENV['S3_DOMAIN_STYLE'] correctly as ':s3_path_url', however it is NOT setting the correct value in the Paperclip initializer (even after restarting the rails service):

We are seeing this:
```
irb(main):004:0> ::Paperclip::Attachment.default_options[:url]
=> ":s3_domain_url"
```

Rather than this:
```
irb(main):004:0> ::Paperclip::Attachment.default_options[:url]
=> ":s3_path_url"
```

Will make more revisions.

**Comment by JackBracken** [comment]
Hi, I was wondering if there was any progress on this? I've looked into it myself and I have a suspicion that Spring is caching your ENV variables, so you're not seeing them change.

For example, when I add `Spring.watch('../.env')` to `config/spring.rb` and set `S3_DOMAIN_STYLE` in my `.env` file the changes are picked up.

```
$ bundle exec rails c
Loading development environment (Rails 4.1.14.2)
[1] pry(main)> ::Paperclip::Attachment.default_options[:url]
=> ":s3_path_url"
[2] pry(main)> quit

$ vim .env
# Make my changes

$ bundle exec rails c
Loading development environment (Rails 4.1.14.2)
[1] pry(main)> ::Paperclip::Attachment.default_options[:url]
=> ":s3_domain_url"
[2] pry(main)> quit


```

**Comment by robbkidd** [comment]
@JackBracken This PR is paused for a bit. It may conjoin with #1411 which has some updates to configuring Paperclip and S3. Thanks for the tip on Spring. I can confirm your finding that it is Spring that is the culprit of the unread environment variable change. As you noted, configuring Spring to monitor the `.env` files seems to solve it by telling Spring to reload the app when those files are updated.

```diff
diff --git a/src/supermarket/config/spring.rb b/src/supermarket/config/spring.rb
new file mode 100644
index 0000000..9b74b2b
--- /dev/null
+++ b/src/supermarket/config/spring.rb
@@ -0,0 +1,2 @@
+Spring.watch '../.env'
+Spring.watch '../.env.development'
```

```shell
> spring status
Spring is running:

 5486 spring server | supermarket | started 7 mins ago
 5487 spring app    | supermarket | started 7 mins ago | development mode

> vim .env
# make the changes

> spring status
Spring is running:

 5486 spring server | supermarket | started 7 mins ago
 5879 spring app    | supermarket | started 10 secs ago | development mode
```

Because this is Spring-related, I expect [this problem only exists in development environments](https://github.com/chef/supermarket/blob/master/src/supermarket/Gemfile#L72-L73) and a `supermarket-ctl restart rails` _would_ have picked up the environment change in production.

**Comment by JackBracken** [comment]
Thanks for the update @robbkidd, the hard work you guys do is appreciated ;)

**Comment by robbkidd** [comment]
I poked a little bit at this branch. In a working copy test branch, I merged these changes into the navmenu-hiding PR's branch where paperclip got an upgrade. I had to twiddle the Spring config a little to get it to monitor `.env.development` and `.env.test` for environment changes per the [previous](https://github.com/chef/supermarket/pull/1307#issuecomment-254176045) [comments](https://github.com/chef/supermarket/pull/1307#issuecomment-254215815).

✅  In an omnibus install, the default for `::Paperclip::Attachment.default_options[:url]` of  `':s3_domain_url'` gets set and is overridable with the attribute.

⚠️  `::Paperclip::Attachment.default_options[:url]` is `nil` in dev/test if S3 is enabled (by setting four S3 environment variables) and if `S3_DOMAIN_STYLE` is _not_ set by a human somehow (say by putting it in the `.env{.development|.test}` file).

**Comment by robbkidd** [comment]
I have rebased #1411 to start with this PR's commit. I aim to approve and merge #1411 soon which will bring this PR into the fold and close it.

**Pull request merged**